performance_regression_cdc_test.py

```diff
@@ -0,0 +1,294 @@
+            time.sleep(60)
+
+        node1.run_cqlsh(f"ALTER TABLE {self.keyspace}.{self.table} WITH cdc = {{'enabled': true}}")
+
+        _ = self._workload_cdc(write_cmd, 2,  # pylint: disable=unused-variable
+                               test_name="test_write",
+                               sub_type="cdc_enabled")
+
+        self.wait_no_compactions_running()
+        self.run_fstrim_on_all_db_nodes()
+
+        try:
+            node1.run_cqlsh(f"TRUNCATE TABLE {self.keyspace}.{self.table}", timeout=300)
+        except (UnexpectedExit, Failure) as details:
+            self.log.warning("Trancate error %s. Sleep and continue", details)
+            time.sleep(60)
+        try:
+            node1.run_cqlsh(f"TRUNCATE TABLE {self.keyspace}.{self.table}_scylla_cdc_log", timeout=300)
+        except (UnexpectedExit, Failure) as details:
+            self.log.warning("Trancate error %s. Sleep and continue", details)
+            time.sleep(60)
+
+        node1.run_cqlsh(f"ALTER TABLE {self.keyspace}.{self.table} WITH cdc = {{'enabled': true, 'preimage': true}}")
+
+        _ = self._workload_cdc(write_cmd, 2,  # pylint: disable=unused-variable
+                               test_name="test_write",
+                               sub_type="cdc_preimage_enabled")
+
+        self.wait_no_compactions_running()
+
+        self.check_regression_with_baseline(base_line="cdc_disabled")
+
+    def test_write(self):  # pylint: disable=unused-variable
+        self.keyspace = "keyspace1"
+        self.table = "standard1"
+        write_cmd = self.params.get("stress_cmd_w")
+
+        _ = self._workload_cdc(write_cmd, 2,  # pylint: disable=unused-variable
+                               test_name="test_write",
+                               sub_type="cdc_disabled")
+
+        node1: BaseNode = self.db_cluster.nodes[0]
+        self.wait_no_compactions_running()
+        self.run_fstrim_on_all_db_nodes()
+        try:
+            node1.run_cqlsh(f"TRUNCATE TABLE {self.keyspace}.{self.table}", timeout=300)
+        except (UnexpectedExit, Failure) as details:
+            self.log.warning("Trancate error %s. Sleep and continue", details)
+            time.sleep(60)
+
+        node1.run_cqlsh(f"ALTER TABLE {self.keyspace}.{self.table} WITH cdc = {{'enabled': true}}")
+
+        _ = self._workload_cdc(write_cmd, 2,  # pylint: disable=unused-variable
+                               test_name="test_write",
+                               sub_type="cdc_enabled")
+
+        self.wait_no_compactions_running()
+        self.run_fstrim_on_all_db_nodes()
+
+        try:
+            node1.run_cqlsh(f"TRUNCATE TABLE {self.keyspace}.{self.table}", timeout=300)
+        except (UnexpectedExit, Failure) as details:
+            self.log.warning("Trancate error %s. Sleep and continue", details)
+            time.sleep(60)
+        try:
+            node1.run_cqlsh(f"TRUNCATE TABLE {self.keyspace}.{self.table}_scylla_cdc_log", timeout=300)
```

Comment:
You can add a helper function to truncate table, and catch the exception inside it, there are some repeated code for truncate tables.

Reply:
Good catch. thanks.

Reply:
fixed